source/ast/symbols/MemberSymbols.cpp

```diff
@@ -2214,34 +2213,89 @@ struct NetAliasVisitor {
     const ASTContext& context;
     const NetType* commonNetType = nullptr;
     bool issuedError = false;
+    // To collect aliased bits info
+    AliasMap aliasedBits;
 
     NetAliasVisitor(const ASTContext& context) : context(context) {}
 
+    template<typename T>
+    const Symbol* processExpr(const T& expr) {
+        if (auto sym = expr.getSymbolReference()) {
+            if (sym->kind != SymbolKind::Net) {
+                context.addDiag(diag::NetAliasNotANet, expr.sourceRange) << sym->name;
+            }
+            else {
+                auto& nt = sym->template as<NetSymbol>().netType;
+                if (!commonNetType) {
+                    commonNetType = &nt;
+                }
+                else if (commonNetType != &nt && !issuedError) {
+                    auto& diag = context.addDiag(diag::NetAliasCommonNetType, expr.sourceRange);
+                    diag << sym->name;
+                    diag << nt.name << commonNetType->name;
+                    issuedError = true;
+                }
+            }
+            return sym;
+        }
+        return nullptr;
+    }
+
     template<typename T>
     void visit(const T& expr) {
+        auto toInt = [](const ConstantValue& cV) {
+            return !cV.bad() && cV.isInteger() ? cV.integer().as<int64_t>() : std::nullopt;
+        };
+
         if constexpr (std::is_base_of_v<Expression, T>) {
             switch (expr.kind) {
                 case ExpressionKind::NamedValue:
-                case ExpressionKind::MemberAccess:
-                case ExpressionKind::ElementSelect:
+                case ExpressionKind::MemberAccess: {
+                    if (const auto* sym = processExpr(expr)) {
+                        aliasedBits[sym] = {
+                            AliasedBits{std::nullopt, std::nullopt, expr.sourceRange}};
+                    }
+                    break;
+                }
+                case ExpressionKind::ElementSelect: {
+                    if (const auto* sym = processExpr(expr)) {
+                        const auto& eSelExpr = expr.template as<ElementSelectExpression>();
+                        // If value can't be evaluated that means it is unbounded
+                        const auto evalVal = toInt(context.tryEval(eSelExpr.selector()));
```

Comment:
I don't think this is correct in all cases. An example is when there is multiple selections of a multi-dimensional array. It's actually pretty tricky to figure out the statically assigned bits of an arbitrary expression -- that's essentially what the ValueDriver code in ValueSymbol.cpp is doing to be able to issue multi-driven errors. It occurs to me that it may make more sense to use that code to implement the net alias check. You could add a new flag to AssignFlags that says it's a net alias and do the check for overlap there.